### PR TITLE
Adds Filter Exclusion Name character limit & related UI

### PR DESCRIPTION
**Pull request by bomanimc**
This PR address issue #13140 by doing the following:
1) Truncates exclusion names over 20 characters.
2) Adds text underneath the input bar to note the remaining characters.

screen shot 2017-03-06 at 8 09 04 pm

Note: This has one string (for the "characters remaining" text) that is externalized in nls/root.

#### Questions: 
I'm having trouble unit testing this, which I've left out of this PR for now. This is my first time working on the project, and I believe I'm making a mistake causing my unit tests to not be considered when I try to run them from the testing interface. None of the `console.log` statements I've added to my unit test pass through. My process has been: 
1) Add a unit test in FileFilters-test.js
2) Reload Brackets
3) Test the FileFilters unit test section.

### CODE REVIEW

File: src/search/FileFilters.js [hunk 1]
```diff
@@ -337,10 +344,24 @@ define(function (require, exports, module) {
...
         }
 
+        $nameField.bind('input', function () {
+            var remainingCharacters = FILTER_NAME_CHARACTER_MAX - $(this).val().length;
+            $remainingField.text(StringUtils.format(
+                Strings.FILTER_NAME_REMAINING,
+                remainingCharacters
+            ));
+
+            if (remainingCharacters < 0) {
```
Comment (zaggino) [inline src/search/FileFilters.js]: unless i'm at lets say 3/4 of `FILTER_NAME_CHARACTER_MAX`, $remainingField could be hidden?

File: src/search/FileFilters.js [hunk 2]
```diff
@@ -337,10 +344,24 @@ define(function (require, exports, module) {
...
+            } else {
+               $remainingField.removeClass("exclusions-name-characters-limit-reached");
+            }
+        });
+
         dialog.done(function (buttonId) {
             if (buttonId === Dialogs.DIALOG_BTN_OK) {
                 // Update saved filter preference
-                setActiveFilter({ name: $nameField.val(), patterns: getValue() }, index);
+                setActiveFilter({ name: $nameField.val().substr(0, FILTER_NAME_CHARACTER_MAX), patterns: getValue() }, index);
```
Comment (zaggino) [inline src/search/FileFilters.js]: do not do this, `OK` button should be disabled if `$nameField.val()` is more than `FILTER_NAME_CHARACTER_MAX`

**Comment by zaggino** [comment]
Travis says:
```
bomanimc has NOT submitted the contributor license agreement. See http://dev.brackets.io/brackets-contributor-license-agreement.html
```
Please sign the agreement before I can merge the code.

**Comment by bomanimc** [comment]
@zaggino Just signed the form! My bad on that.

**Review by zaggino — APPROVED** [review]
Tested, LGTM

**Comment by zaggino** [comment]
Thanks for the PR @bomanimc , nice work.

**Pull request merged**